management/virtualmachineimage/client.go

```diff
@@ -19,10 +20,33 @@ func NewClient(client management.Client) Client {
 	return Client{client}
 }
 
-func (c Client) ListVirtualMachineImages() (ListVirtualMachineImagesResponse, error) {
+//ListVirtualMachineImages lists the available VM images, filtered by the optional parameters.
+//See https://msdn.microsoft.com/en-us/library/azure/dn499770.aspx
+func (c Client) ListVirtualMachineImages(parameters *ListParameters) (ListVirtualMachineImagesResponse, error) {
```

Comment:
any reasons for this to be a pointer? I remember in the rest of the repo (at least in storage/) we're following zero-value pattern rather than passing nilptrs.